server/app/Main.hs

```diff
@@ -1,4 +1,46 @@
 import Lib
+import Options (Option (..), runOptionParser)
+import qualified Options.Applicative as Opt
+
+getConfig :: IO ServerConfig
+getConfig = runOptionParser $ \envOption ->
+  ServerConfig
+    <$> envOption
+      ( Option
+          { optHelp = "Path to the nix program database.",
+            optMetaVar = "PATH",
+            optLong = "program-db",
+            optShort = 'd',
+            optEnvKey = "NIX_PROGRAM_DB",
```

Comment:
For envvar names, I suggest `<APP_PREFIX>_<THING>` style.
Unless so, it easily go hard to track which env var is consumed by which application.

For this one, `BINPLZ_PROGRAM_DB` looks better.